### PR TITLE
COMMERCE-618

### PR DESCRIPTION
**Pull request by ZoltanTakacs**
https://issues.liferay.com/browse/COMMERCE-618

Resend of https://github.com/brianchandotcom/liferay-portal/pull/62865

Hi Brian,

**tl;dr**
I tried to avoid commiting the cached resources here, so I've put up a gradle script which can download the necessary 3rd party dependencies for the installer.

**In details:** 
Talend Open Studio uses a mechanism to fetch the actual plugin's dependencies, but their Nexus repository is not so reliable. There were several times when a dependency lib (Talend related ones) just was removed from their server and on another day it was popped up again which result unpredictable success of installation of the Liferay component.

Talend stores these dependencies in its local maven repository and what I've done is installed a fresh bundle of TOS used those Maven deps as a baseline and installed our component wait until Talend fetches / stores all the dependencies and compared the two states. This result can be found in https://github.com/ZoltanTakacs/liferay-binaries-cache-2017/tree/talend-installer-3rd-party-cache
which is used by our installer script to copy them directly to TOS's Maven local so after the component installation everything will be ready and the component can be used immediately.

Also there is no way in the Open studio to set the repository where the libs can be downloaded by default. I asked them on their community forum how to solve this, but it's only available in the subscription version of Talend:
* [Auto downloading external modules for a custom component in Open Studio](https://community.talend.com/t5/Design-and-Development/Auto-downloading-external-modules-for-a-custom-component-in-Open/m-p/48652/highlight/true#M15434)

Without the installer like a nightmare to make this component work, unless we convince them to bundle it up in their next version of TOS, like the Salesforce, Magento etc.. ones.. :(

Please let me know what do you think about it.

Thanks,
Zoli

### CODE REVIEW

**Comment by brianchandotcom** [comment]
Very very interesting @ZoltanTakacs (sorry took so long to review, I want to do this right as this is our first Go module, and I see us doing more Go stuff in the future, so I wanted to do this one right).

1.) modules/etl/talend/installer/vendor/github.com/go-ini/ini
modules/etl/talend/installer/vendor/github.com/mitchellh/gox
modules/etl/talend/installer/vendor/github.com/otiai10/copy

Are those accidentally checked in? They all have similar error messages of:

Submodule repository 'modules/etl/talend/installer/vendor/github.com/otiai10/copy' not found.

2.) Regarding

modules/etl/talend/installer/build.gradle

https://github.com/ZoltanTakacs/liferay-binaries-cache-2017/archive/talend-installer-3rd-party-cache.zip

Right now, that file is "magic". I want you to make another module in "modules/third-party/com-talend-installer-dependencies"

I will then publish that module as "com.liferay" "com.talend.thirdparty.dependencies"

you can then depend on it in your actual talend module.

See https://github.com/liferay/com-liferay-osb-asah-private/blob/7.0.x-private/osb-asah-cerebro-extractor/build.gradle It uses a ".gz" file.

Then we won't need another download plugin.

Sorry for taking so long, I'll review your next pull asap, but it should just be to modules/third-party.

I don't like hiding things in magic.

Thx!
